### ISSUE TITLE
ga-wdi-boston/node-api-promises

### ISSUE THREAD

**Issue by jrhorn424** [ga-wdi-boston/node-api-promises#1]

**Comment by jrhorn424** [comment]
147 minute actual

**Comment by jrhorn424** [comment]
I think this deserves a 3 estimate in the future. I think the lesson went well, and I believe that since Promises can be a bit tough to wrap your head around at first, spending this much time practicing was important for the lessons that came afterward.

**Comment by gaand** [comment]
130 but only because I didn't time the lab.

**Comment by payne-chris-r** [comment]
Ok. This one became a bit of a `can-o-worms` because people felt like they "understood promises because we used them with ajax" and they knew how to _use_ promises, but they don't understand the mechanics of promises. Actual was more like 4 units (~3.5 => 9:40-12, 1-1:30). May need more attention or a smaller build up of an example. See #14

**Comment by jrhorn424** [comment]
> they don't understand the mechanics of promises

I'm going to take a radical stance and say argue that they don't need to. Understanding the contract is enough. I don't 100% believe that myself, but we should have a discussion about it.

Understanding the mechanics was more important when we didn't have Promises in mongoose and when we were still using jQuery 2.

Do they have to promisify anything anymore?

cc @gaand

**Comment by payne-chris-r** [comment]
No, but understanding promises helps you understand promisified ember objects and arrays... So, IMO understanding them is reeeeeally helpful.

**Comment by raq929** [comment]
There were a lot of questions about promises, so we didn't get to the last lab, but I think that was OK.
